storage: panic cloning roachpb.Error during lease acquisition

Saw this panic on `delta` while it was running #10867.

```
panic: attempt to clone range 72 was not found, which contains 0d5e20b0-fb94-4fb2-ab9c-eae7c7099e15 [recovered]
```

The `range 72 was not found` sounds like a `roachp.RangeNotFoundError`. Not sure how it could contain that UUID. The stack trace is below.

```
panic(0x16f07e0, 0xc449552680)
	/usr/local/go/src/runtime/panic.go:458 +0x243
github.com/cockroachdb/cockroach/pkg/util/protoutil.Clone(0x23fd860, 0xc42ba323c0, 0xc4233a68d8, 0x1488d4772c2167f9)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/protoutil/clone.go:60 +0x247
github.com/cockroachdb/cockroach/pkg/storage.(*pendingLeaseRequest).InitOrJoinRequest.func1(0x7f5b2dff3f80, 0xc42e98e810)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_range_lease.go:154 +0x359
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1(0xc420096000, 0x20eb9dc, 0x1e, 0x9f, 0x0, 0x0, 0xc42e98e720, 0x7f5b2dfe8148, 0xc42000e280)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:264 +0xdf
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:265 +0x234
```

Cc @tamird who knows the proto cloning code and @andreimatei who knows the lease code

--- a/pkg/storage/replica_range_lease.go
+++ b/pkg/storage/replica_range_lease.go
@@ -25,7 +25,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
-	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
 	"github.com/pkg/errors"
 )
 
@@ -143,15 +142,14 @@ func (p *pendingLeaseRequest) InitOrJoinRequest(
 		// Send result of lease to all waiter channels.
 		replica.mu.Lock()
 		defer replica.mu.Unlock()
-		for i, llChan := range p.llChans {
-			// Don't send the same pErr object twice; this can lead to races. We could
-			// clone every time but it's more efficient to send pErr itself to one of
-			// the channels (the last one; if we send it earlier the race can still
-			// happen).
-			if i == len(p.llChans)-1 {
-				llChan <- pErr
+		for _, llChan := range p.llChans {
+			// Don't send the same transaction object twice; this can lead to races.
+			if pErr != nil {
+				pErrClone := *pErr
+				pErrClone.SetTxn(pErr.GetTxn())
+				llChan <- &pErrClone
 			} else {
-				llChan <- protoutil.Clone(pErr).(*roachpb.Error) // works with `nil`
+				llChan <- nil
 			}
 		}
 		p.llChans = p.llChans[:0]

--- a/pkg/storage/replica_test.go
+++ b/pkg/storage/replica_test.go
@@ -1735,15 +1735,6 @@ func TestLeaseConcurrent(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	const num = 5
 
-	// The test was written to test this individual block below. The worry
-	// was that this would NPE (it does not; proto.Clone is unusual in that it
-	// returns the input value instead).
-	{
-		if protoutil.Clone((*roachpb.Error)(nil)).(*roachpb.Error) != nil {
-			t.Fatal("could not clone nil *Error")
-		}
-	}
-
 	// Testing concurrent range lease requests is still a good idea. We check
 	// that they work and clone *Error, which prevents regression of #6111.
 	const origMsg = "boom"

